modules/transactionpool/accept.go

```diff
@@ -85,51 +88,32 @@ func relatedObjectIDs(ts []types.Transaction) []ObjectID {
 	return oids
 }
 
-// checkMinerFees checks that the total amount of transaction fees in the
-// transaction set is sufficient to earn a spot in the transaction pool.
-func (tp *TransactionPool) checkMinerFees(ts []types.Transaction) error {
-	// Transactions cannot be added after the TransactionPoolSizeLimit has been
-	// hit.
-	if tp.transactionListSize > TransactionPoolSizeLimit {
-		return errFullTransactionPool
-	}
-
-	// The first TransactionPoolSizeForFee transactions do not need fees.
-	if tp.transactionListSize > TransactionPoolSizeForFee {
-		// Currently required fees are set on a per-transaction basis. 2 coins
-		// are required per transaction if the free-fee limit has been reached,
-		// adding a larger fee is not useful.
-		var feeSum types.Currency
-		for i := range ts {
-			for _, fee := range ts[i].MinerFees {
-				feeSum = feeSum.Add(fee)
-			}
-		}
-		feeRequired := TransactionMinFee.Mul64(uint64(len(ts)))
-		if feeSum.Cmp(feeRequired) < 0 {
-			return errLowMinerFees
-		}
+// requiredFeesToExtendTpool returns the amount of fees required to extend the
+// transaction pool to fit another transaction set. The amount returned has the
+// unit 'currency per byte'.
+func (tp *TransactionPool) requiredFeesToExtendTpool() types.Currency {
+	// If the transaction pool is nearly empty, it can be extended even if there
+	// are no fees.
+	if tp.transactionListSize < TransactionPoolSizeForFee {
+		return types.ZeroCurrency
 	}
-	return nil
+
+	// Calculate the fee required to bump out the size of the transaction pool.
+	ratioToTarget := float64(tp.transactionListSize) / TransactionPoolSizeTarget
+	feeFactor := math.Pow(ratioToTarget, TransactionPoolExponentiation)
+	return types.SiacoinPrecision.MulFloat(feeFactor).Div64(1000)
```

Comment:
The `Div(1000)` is a bit magic; maybe clearer with a comment and separate var:
```go
targetFee := types.SiacoinPrecision.Div64(1000) // 1 SC / KB
return targetFee.MulFloat(feeFactor)
```
not a blocking issue though